rs/messaging/src/routing/stream_handler/tests.rs

```diff
@@ -83,520 +90,390 @@ fn oversized_reject_message_is_truncated() {
     );
 }
 
+/// Tests that inducting a loopback stream where `signals_end` < `end` panics (this must be equal
+/// for the loopback stream, because it is its own reverse stream).
 #[test]
 #[should_panic(
     expected = "Invalid message indices in stream slice from subnet g24bn-xymaa-aaaaa-aaaap-yai: messages begin (21) != stream signals_end (22)"
 )]
-fn induct_loopback_stream_with_signals_panics() {
-    with_test_replica_logger(|log| {
-        let (stream_handler, mut state, _) = new_fixture(&log);
-
-        // A loopback stream containing 2 messages and 1 signal.
-        let loopback_stream = generate_loopback_stream(StreamConfig {
-            messages_begin: 21,
-            message_count: 2,
+fn induct_loopback_stream_with_signals_end_before_messages_end_panics() {
```

Comment:
Nit: I find the earlier "loopback stream with signals panics" description a lot easier to grasp than "`signals_end < end` (this must be equal...".

Feel free to add a doc comment explaining why we expect no signals in the loopback stream, but keep it at a high level (i.e. "no signals") instead of putting it in terms of the implementation ("this fields < that field because ...").

Reply:
Changed.

Reply:
I meant putting the test case name and documentation in terms of "a loopback stream with signals panics" instead of an implementation-specific and not very obvious "this index before that index panics".

I.e. something along the lines of:
```rust
/// Tests that inducting a loopback stream with signals panics: `induct_loopback_stream()`
/// always consumes all\ produced signals.
#[test]
#[should_panic(
    expected = "Invalid message indices in stream slice from subnet g24bn-xymaa-aaaaa-aaaap-yai: messages begin (21) != stream signals_end (22)"
)]
fn induct_loopback_stream_with_signals_panics() {
```

(Also, messages being GC-ed up to `sinals_end` and `signals_end` increasing monotonically doesn't really make any difference as to why we expect there to be no signals in the loopback stream.)

Reply:
My mental model was a bit different also considering the test just below this one. However I ditched the second part of your comment since without the knowledge that `induct_loopback_stream()` does produce signals but then immediately gc's them, this seems confusing.